include/seqan3/range/decorator/gap_decorator_anchor_set.hpp

```diff
@@ -783,12 +786,31 @@ class gap_decorator_anchor_set
     }
 
     //!\brief Stores a (copy of a) view to the ungapped, underlying sequence.
-    decltype(view::all(std::declval<inner_type const &>())) ungapped_view{};
+    decltype(view::all(std::declval<inner_type &&>())) ungapped_view{};
 
     //!\brief Set storing the anchor gaps.
     anchor_set_type anchors{};
 };
 
+/*!\name Type deduction guides
+ * \{
+ */
+//!\brief Ranges (non views!) always deduce to `const & range_type` since they are access-only anyway.
```

Comment:
```suggestion
//!\brief Ranges (not views!) always deduce to `const & range_type` since they are access-only anyway.
```

Reply:
This is for PR #777 and will be rebased after it is merged